R/examine-constraints.R

```diff
@@ -33,13 +33,13 @@ dm_examine_constraints <- function(dm, progress = NA) {
   dm %>%
     dm_examine_constraints_impl(progress = progress) %>%
     rename(columns = column) %>%
-    mutate(columns = new_keys(columns)) %>%
+    mutate(columns = new_keys(columns), repair_plan = NULL) %>%
     new_dm_examine_constraints()
 }
 
-dm_examine_constraints_impl <- function(dm, progress = NA) {
+dm_examine_constraints_impl <- function(dm, progress = NA, fk_repair = "none") {
```

Comment:
```suggestion
dm_examine_constraints_impl <- function(dm, progress = NA, fk_repair = NULL, sample = TRUE) {
```

Could we use two arguments:

- `fk_repair` defines what kind of repair we consider: adding dimensions or pruning facts. Are there other options?
- `sample` defines if we're satisfied with a sample (current implementation) or if we want all instances (for cleanup)